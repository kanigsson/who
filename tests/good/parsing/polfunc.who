logic fst ['a 'b||] : 'a * 'b -> 'a
logic snd ['a 'b||] : 'a * 'b -> 'b
let f [||'e] (g : int -> {'e} int) =
  { fst g 5 cur }
    g 5
  { r : snd g 5 old cur r }

