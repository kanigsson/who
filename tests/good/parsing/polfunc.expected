logic fst ['a 'b||] : (a * b) -> a
logic snd ['a1 'b1||] : (a1 * b1) -> b1
let
  f [||'e] = (fun (g:int ->{ 'e } int) -> {fst g 5 cur} g 5
             {r : snd g 5 old cur1 r}) 
